### PR TITLE
Move stuff from AttackableUnit to ObjAiBase

### PR DESCRIPTION
**Pull request by moonshadow565**
Stuff that is moved:
-buff add,remove,update ...
-character data loading
-aadata
-everything for autoattacks
-CC
-waypoint refresh, stop movement, movement handling

Removed:
-CC invulnerability (this needs replication changes to work, so until  those land its removed)
-unit distress  (too complex to fix it, can be added latter)

Depends on: https://github.com/LeagueSandbox/LeagueSandbox-Default/pull/55

### CODE REVIEW

File: GameServerLib/Logic/GameObjects/AttackableUnits/AI/ObjAIBase.cs [hunk 1]
```diff
@@ -12,22 +17,71 @@ public class ObjAIBase : AttackableUnit
         private object BuffsLock { get; }
         private Dictionary<string, Buff> Buffs { get; }
         
+        private List<UnitCrowdControl> crowdControlList = new List<UnitCrowdControl>();
```
Comment (FurkanS1821) [inline GameServerLib/Logic/GameObjects/AttackableUnits/AI/ObjAIBase.cs:20]: `_crowdControlList`

File: GameServerLib/Logic/GameObjects/Missiles/Projectile.cs [hunk 2]
```diff
@@ -152,7 +152,8 @@ protected virtual void CheckFlagsForUnit(AttackableUnit unit)
                     }
                     else
                     { // auto attack
-                        Owner.AutoAttackHit(u);
+                        if(u is ObjAIBase ai)
+                            ai.AutoAttackHit(u);
```
Comment (FurkanS1821) [inline GameServerLib/Logic/GameObjects/Missiles/Projectile.cs]: brackets please

File: GameServerLib/Logic/ObjectManager.cs [hunk 3]
```diff
@@ -255,12 +255,14 @@ public void StopTargeting(AttackableUnit target)
                     var u = kv.Value as AttackableUnit;
                     if (u == null)
                         continue;
-
-                    if (u.TargetUnit == target)
+                    if(u is ObjAIBase ai)
```
Comment (FurkanS1821) [inline GameServerLib/Logic/ObjectManager.cs]: space please

File: GameServerLib/Logic/GameObjects/Missiles/Projectile.cs [hunk 4]
```diff
@@ -152,7 +152,8 @@ protected virtual void CheckFlagsForUnit(AttackableUnit unit)
                     }
                     else
                     { // auto attack
-                        Owner.AutoAttackHit(u);
+                        if(u is ObjAIBase ai)
```
Comment (FurkanS1821) [inline GameServerLib/Logic/GameObjects/Missiles/Projectile.cs]: space please

**Review by FurkanS1821 — CHANGES_REQUESTED** [review]
Seems alright to me, except *very few* guideline stuff.

File: GameServerLib/Logic/GameObjects/AttackableUnits/AI/ObjAIBase.cs (hunk 1 above)
Comment (moonshadow565) [reply GameServerLib/Logic/GameObjects/AttackableUnits/AI/ObjAIBase.cs:20]: It was ``crowdControlList`` already in ``AttackableUnit``.
Considering i would like to keep changes to only moving stuff from AttackablrUnit, i don't think this is a good idea.
Not to mention the whole UnitCrowdControl thing mimics ActionState flags + invulnerability and targetability.
It should be addressed in another PR.

File: GameServerLib/Logic/GameObjects/Missiles/Projectile.cs (hunk 4 above)
Comment (moonshadow565) [reply GameServerLib/Logic/GameObjects/Missiles/Projectile.cs]: please explain(newline maybe?)

File: GameServerLib/Logic/ObjectManager.cs (hunk 3 above)
Comment (moonshadow565) [reply GameServerLib/Logic/ObjectManager.cs]: pleaae explain(newline maybe?)

Comment (Deudly) [reply GameServerLib/Logic/ObjectManager.cs]: Space between `if` and `(u...`

File: GameServerLib/Logic/GameObjects/Missiles/Projectile.cs (hunk 4 above)
Comment (Deudly) [reply GameServerLib/Logic/GameObjects/Missiles/Projectile.cs]: Space between `if` and `(u...`

**Comment by MythicManiac** [comment]
There's some styleguide 🚲-shedding (which honestely we should already automate), and the biggest issue I can see is this not compiling.

**Review by FurkanS1821 — COMMENTED** [review]
Fix the commit message and this is ready to go.

**Review by FurkanS1821 — APPROVED** [review]

**Pull request merged**